jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -249,7 +238,6 @@ def _bg_subset_selected_changed(self, event={}):
 
     def vue_do_aper_phot(self, *args, **kwargs):
         if self._selected_data is None or self._selected_subset is None:
-            self.reset_results()

```

Comment:
I still think you need to reset when new calculation is requested.

Reply:
can you show a case where that is needed?  I think everything just gets overwritten for a successful run and the code is redundant, but I'm not 100% sure.

Reply:
The point is, when this button is pressed, things should be cleared and only show new calculated results. If there is any failure at all (this, or the `except Exception as e` below), no numbers should show, to make it clear that something has gone wrong.

The logic here should not affect the issue you are trying to fix. Deleting a Subset should not trigger this method.

Or am I misunderstanding the situtation?

Reply:
Deleting all subsets *does* trigger resetting the results and ends up hiding both the most recent results and the entire table.  I can move the entire table outside of that logic and to its own logic (just checking the length of the table or something), but that feels like extraneous logic and inconsistent behavior to me.


https://user-images.githubusercontent.com/877591/227565965-1c1781c2-2ef7-463e-aa68-87d10ad824e3.mov

Reply:
Delete subset event triggers `vue_do_aper_phot`? 🤯

Reply:
No, but the subset selection itself becomes blank, and then `_subset_selected_changed` clears the results.

Reply:
Should all the previous results (including the table) disappear?  What if instead we show a message that the previous run failed above all of those so they are not lost?

Reply:
> Ok, so what if we show the error message in the UI in place of the results and keep the table?

This might be a good enough compromise.